### ISSUE TITLE
Specifying samples for training in bcftools roh

### ISSUE THREAD

**Issue by gtamazian** [samtools/bcftools#276]
In **bcftools roh**, the HMM model for detecting runs of homozygosity is trained using all samples in the specified VCF file. However, a VCF file may contain samples from a number of populations and it would be convenient to specify samples for the model training. 

For example, one could introduce **bcftools roh** command-line options for it:

```
HMM options:
     ...
     -p, --population [^]<list>       comma-separated list of samples for HMM training
     -P, --population-file [^]<file>  file of samples to include (or exclude with "^" prefix) for HMM training 
```

**Comment by pd3** [comment]
What version of `bcftools` is that? I believe all recent versions do the training using the single `-s` sample.

**Comment by gtamazian** [comment]
I just discovered that one may specify multiple comma-separated samples in the `-s` option. However, I need to check how the HMM training is carried out.

**Comment by gtamazian** [comment]
@pd3 Thanks for your remark. I checked the repository's history and noticed that **bcftools roh** had been modified significantly from its first version.

I have been studying homozygosity analysis that was carried out with **bcftools roh** in one genome paper published recently. The author who was responsible for that warned me that a VCF file of original variants had to be split into separate files, one for each subpopulation, to apply **bcftools roh** to them. The reason was that the version they used trained the HMM using variants for *all* samples in the given VCF file. I suppose it was one of the earlier versions of the program.

**Comment by pd3** [comment]
Hmm, I am not sure that ever was the case. Viterbi training was introduced in b1b2fb20 and this already used one sample only. Maybe you mean the estimate of allele frequencies, which had to be supplied in a separate file or as an INFO tag in early versions, then `-s` could be used to subset the samples.  Anyway, if the current version does not work as described in the documentation, just let me know.

**Comment by pd3** [comment]
There was a bug in estimating allele frequencies with `-e -`. This should be fixed now in 3ba5f38ced4e3938348fa231a2f3d794ee43bc5a. I hope this resolves your question, if not, please reopen.
